### PR TITLE
[agent] Add support for running unprivileged

### PR DESCRIPTION
**Pull request by jawnsy**
Run as a non-root user with a read-only filesystem, mounting an emptyDir volume for temporary state files. This also prevents privilege escalation by default.

### CODE REVIEW

**Review by jamiezieziula — DISMISSED** [review]
yeehaw this is awesome 🤠

**Comment by jawnsy** [comment]
@jamiezieziula I'll wait until you merge yours, then will merge this to main 🥳 

In the meantime, I'll just leave this in draft

**Review by jamiezieziula — DISMISSED** [review]

**Review by jamiezieziula — APPROVED** [review]

**Pull request merged**
